client/src/app/shared/shared-main/video/video.service.ts

```diff
@@ -398,9 +398,14 @@ export class VideoService implements VideosProvider {
       }
     ]
 
-    return base
+    const videoPrivacies = base
       .filter(o => !!privacies.find(p => p.id === o.id)) // filter down to privacies that where in the input
       .map(o => ({ ...privacies[o.id - 1], ...o })) // merge the input privacies that contain a label, and extend them with a description
+
+    return {
+      defaultPrivacyId: videoPrivacies.find(p => p.id === defaultPrivacyId).id || videoPrivacies[0].id,
```

Comment:
Have you tested this code? Because `videoPrivacies.find` can return null

Reply:
I corrected the attribute access.